transformer_engine/pytorch/csrc/userbuffers/userbuffers-host.cpp

```diff
@@ -279,8 +282,12 @@ int create_communicator_grouped2(communicator **comm, int pipegpus, int pipenode
                            (NVTE_MAX_SMS + 100) * sizeof(int)));
   for (int i = 0; i < 100 + NVTE_MAX_SMS; i++)
     (*comm)->hostflags[i] = 0;
+#if (defined(__arm__) && __ARM_ARCH >= 7) || defined(__arm64__) || defined(__aarch64__)
```

Comment:
Based on my googling, it doesn't seem that `__arm64__` is used by GCC or Clang: 
```suggestion
#if (defined(__arm__) && __ARM_ARCH >= 7) || defined(__aarch64__)
```
For future reference: the DMB instruction was introduced in ARMv7. https://developer.arm.com/documentation/dui0489/c/arm-and-thumb-instructions/miscellaneous-instructions/dmb--dsb--and-isb